barretenberg/cpp/src/barretenberg/polynomials/polynomial.cpp

```diff
@@ -318,18 +325,32 @@ template <typename Fr> Polynomial<Fr>& Polynomial<Fr>::operator*=(const Fr scali
     return *this;
 }
 
-template <typename Fr> Polynomial<Fr> Polynomial<Fr>::full() const
+template <typename Fr> Polynomial<Fr> Polynomial<Fr>::expand(size_t new_start_index, size_t new_end_index) const
```

Comment:
new expand function that can expand a poly's start and end index

Reply:
It's funny because I originally had this and then was asked to make it .full() :)

Reply:
oh lol. I still kept full around though :)

Reply:
no he's still using the `full` interface, he just added this expand in the background right?